design/providerid_format_change.md

```diff
@@ -0,0 +1,128 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# providerid_format_change
+
+## Status
+
+provisional
+
+## Summary
+
+The providerID is generated using Baremetalhost's uuid. 
+While the Baremetalhost's uuid is tied to the life a server, the providerID 
+is related to a provisioning a kubernetes cluster. The current providerID has 
+some limitations due to content and format.
+
+The providerID uses Baremetalhost's uuid, making it static for the life of 
+a server. There is also an issue with it using ```UUID``` that is not known 
+in advance, requiring using complex templates.
+
+## Motivation
+
+Currently, providerID has the `metal3://<bmh-uuid>` format. This value static 
+for the life cycle of the server. Kubernetes deployments and subsequent 
+upgrades of uses the same providerID. This has caused issues with upgrades.
+It could be made to be dynamic by including other resources that are generated
+uniquely for each kubernetes deployment.
+
+The usage of `UUID` creates some challenges as the information is not 
+available in advance. Using resource names would solve to issue.
+
+The current providerID logic makes use of labels in 
+`metal3.io/uuid=<bmh-uuid>` format, also supplied via templates. This does not
+provider extra information as provider-id (`metal3://<bmh-uuid>`) contains the 
+same information. In this regard, the label is redundant.
+
+### Goals
+
+- Add new fields to metal3datatemplate
+- Deprecate usage of node labels.
+- Change providerID format to include dynamic information
+
+### Non-Goals
+
+Making tempates simpler or removing them.
+
+## Proposal
+
+The issues discussed above can be solve by introducing a new field in the 
+metal3datatemplate implicitly.
+
+```
+providerid: "metal3://<namespace>/<bmh name>/<metal3machine name>"
+```
+
+The value is both dynamic and known in advance. Since this is added from the 
+code, there is no need of changing Spec or Status sections of the resource.
+
+Since it is constructed in the code, there is no need for the users to 
+provider provider-id and label via templates.
+
+### User Stories
+
+None
```

Comment:
please add one. As a user, I want my nodes' providerIDs to be unique and automatically generated by CAPM3 ...